src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -28,22 +41,37 @@ const MainDrawerNavigator = props => (
         sceneContainerStyle={styles.navigationSceneContainer}
         edgeWidth={500}
         drawerContent={() => <SidebarScreen />}
+        screenOptions={{
+            cardStyle: styles.navigationScreenCardStyle,
+            headerShown: false,
+        }}
     >
-        <Drawer.Screen
-            name="Report"
-            component={ReportScreen}
-
-            // Providing an empty string here will ensure that the ReportScreen does not show as '/r/undefined'
-            // eslint-disable-next-line react/jsx-props-no-multi-spaces
-            initialParams={{reportID: ''}}
-            options={{
-                cardStyle: styles.navigationScreenCardStyle,
-                headerShown: false,
-            }}
-        />
+        {
+            _.isString(props.initialReportID)
+                ? (
+                    <Drawer.Screen
+                        name="Report"
+                        component={ReportScreen}
+                        initialParams={{reportID: props.initialReportID}}
+                    />
+                )
+                : (
+                    <Drawer.Screen name="loading">
+                        {() => <FullScreenLoadingIndicator visible />}
+                    </Drawer.Screen>
+                )
```

Comment:
Moving the conditional mounting here allows to always have a navigator mounted.

The Report screen will mount once `initialReportID` is available - either instantly the usual case, or after a brief time - login / create account.

In the meantime a placeholder screen is mounted as you can't have a Navigator without screens. It doesn't need to render anything, but I've added the loader. As a name I've used "loading" this would appear as a URL until the Report Screen is mounted, this can be renamed or setup so that it doesn't appear in the URL. You can see this in the video below

I've used the `isString` check in the rare event that no `initialReportId` is not resolved - there's a case where `''` would be set. This is an error case because, as you know initialization logic would always set a `initialReportId` (or navigate to the Concierge conversation). In that case the view would just show the empty report page and you can recover by selecting a different conversation or refreshing the page.

This can perhaps be further improved a bit if state is used. E.g. save state the first time that `props.initialReportID` becomes available and keep the Report Screen mounted ever since that - in case you think that something could delete/remove the `props.initialReportID` e.g. Onyx eviction

##### Video simulating how initialReportId is resolved from a network call
https://user-images.githubusercontent.com/12156624/116292439-610f5c00-a79e-11eb-8283-5bf9ef3c4784.mp4
- You can also see the effect of deleting currently viewed report during runtime

1. Report saved in Onyx is deleted
2. App is reloaded from root `/`
3. The route (report ID) resolves from Network

Reply:
This logic makes sense 👍  
Though I think 

> can be renamed or setup so that it doesn't appear in the URL

we should do this as we don't want a url that has /loading